src/features/ActivityLog/ActivityContext.tsx

```diff
@@ -0,0 +1,113 @@
+import { type PropsWithChildren, createContext, useReducer } from 'react';
+
+import { generateId } from '../../utils/generateId';
+
+import type { Activity, ActivityDetail, User } from './activityLog.interface';
+
+const INITIAL_ACTIVITY_DETAIL: ActivityDetail = {
+  user: {
+    name: '',
+    age: 0,
+    contactNumber: '',
+  },
+  activities: [],
+};
+
+interface IActivityContext {
+  state: ActivityDetail;
+  dispatch: (action: any) => void;
+}
+
+const ActivityContext = createContext<IActivityContext>({
+  state: INITIAL_ACTIVITY_DETAIL,
+  dispatch: () => {},
+});
+
+export enum ActivityActionType {
```

Comment:
If you're using in multiple places, better to have some files for specific constant